src/cache-restore.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright 2023 The MathWorks, Inc.
+
+import * as cache from '@actions/cache';
+import * as core from '@actions/core';
+import * as crypto from "crypto";
+import { State } from './cache-state';
+import { Release } from './matlab';
+
+export async function restoreMATLAB(release: Release, platform: string, architecture: string, products: string[], matlabPath: string): Promise<boolean> {
+    const installHash = crypto.createHash('sha256').update(products.join('|')).digest('hex')
```

Comment:
Should the products list be sorted before creating this hash?

Reply:
Updated